src/features/security-cloud-game/SecurityCloudGame.js

```diff
@@ -1,82 +1,111 @@
-    static propTypes = {
-        /** Width/Height to set the drag and drop clouds to.  Defaults to 64 */
-        cloudSize: PropTypes.number,
-        /** Height to set the game to */
-        height: PropTypes.number.isRequired,
-        /** Function to call when the `DragCloud` is successfuly dropped onto the `DropCloud` */
-        onValidDrop: PropTypes.func,
-        /** Width to set the game to */
-        width: PropTypes.number.isRequired,
-    };
+// pick these numbers to balance accessibility and game complexity
+const CLOUD_SIZE_RATIO = 5;
+const GRID_TRACK_SIZE_RATIO = 20;
+
+const SecurityCloudGame = ({ height, intl: { formatMessage }, onValidDrop, width }) => {
+    const [dropCloudPosition, setDropCloudPosition] = useState(null);
+    const [dragCloudPosition, setDragCloudPosition] = useState(null);
+    const [layout, setLayout] = useState({});
+    const [gameState, setGameState] = useState({
+        isValidDrop: false,
+        isOverlap: false,
+        liveText: '',
+    });
+
+    const messageElementRef = useRef();
+    // to handle resize events
+    const gameBoardSizeRef = useRef({});
+
+    const { cloudSize, gameBoardHeight, gridTrackSize } = layout;
+    const { liveText, isOverlap, isValidDrop } = gameState;
+
+    useLayoutEffect(() => {
+        const { current: messageElement } = messageElementRef;
+        const newGameBoardHeight = height - messageElement.getBoundingClientRect().height;
+        // guardrail to prevent further rendering if the game board height is not positive
+        if (newGameBoardHeight <= 0) return;
+
+        const minGameBoardLength = Math.min(newGameBoardHeight, width);
+        setLayout({
+            gameBoardHeight: newGameBoardHeight,
+            cloudSize: minGameBoardLength / CLOUD_SIZE_RATIO,
+            gridTrackSize: minGameBoardLength / GRID_TRACK_SIZE_RATIO,
+        });
 
-    static defaultProps = {
-        cloudSize: 64,
-    };
+        messageElement.focus();
+    }, [height, width]);
+
+    useEffect(() => {
+        if (!gameBoardHeight) {
+            return;
+        }
+
+        const { height: prevHeight, width: prevWidth } = gameBoardSizeRef.current;
+        const heightRatio = prevHeight ? gameBoardHeight / prevHeight : 1;
+        const widthRatio = prevWidth ? width / prevWidth : 1;
 
-    constructor(props) {
-        super(props);
-        this.occupiedRegions = [];
-        this.state = {
-            isValidDrop: false,
-            isOverlap: false,
-        };
-
-        this.onDrag = throttle(this.onDrag, 100, {
-            leading: true,
-            trailing: true,
+        let newDropCloudPosition;
+        setDropCloudPosition(prevPos => {
+            newDropCloudPosition = !prevPos
+                ? getRandomCloudPosition(cloudSize, gameBoardHeight, width)
+                : { x: prevPos.x * widthRatio, y: prevPos.y * heightRatio };
+            return newDropCloudPosition;
+        });
+        setDragCloudPosition(prevPos => {
```

Comment:
readability for this section is a little difficult but I can't think of much improvements besides adding additional line breaks